sdk/storage/azure-storage-blob/azure/storage/blob/aio/_download_async.py

```diff
@@ -103,14 +114,14 @@ async def _download_chunk(self, chunk_start, chunk_end):
                 check_content_md5=self.validate_content
             )
             try:
-                _, response = await self.client.download(
+                _, response = await cast(Awaitable[Any], self.client.download(
```

Comment:
Another instance of where I reverted our changes in `_download` to have both versions of the `BlobOperations` sync and async. Easier to just shove the output of the code call into the right looking type 😄 
Our attempt ex:
```
_, response = await self.client.download( 
_download_async.py:117: error: Incompatible types in "await" (actual type "Union[Iterator[bytes], Awaitable[AsyncIterator[bytes]]]", expected type "Awaitable[Any]")  [misc]
```
Okay so then I try to explicitly cast it to the async `BlobOperations`

```
_, response = await (cast("BlobOperations", self.client).download(
_download_async.py:117: error: "AsyncIterator[bytes]" object is not iterable  [misc]
```
Great.. the `AsyncIterator[bytes]` which has ITERABLE IN ITS NAME, is not iterable 😭